doc/security/specs/secure_boot/index.md

```diff
@@ -0,0 +1,212 @@
+---
+title: "Secure Boot"
+---
+
+<p style="text-align: right">
+Contributors(s):
+  <a href="https://github.com/moidx">Miguel Osorio</a>,
+  <a href="https://github.com/moidx">Garret Kelly</a>
+</p>
+
+<p style="color: red; text-align: right;">
+  Status: Pre-RFC
+</p>
+
+# Overview
+
+The purpose of secure boot is to ensure that only payloads which have been
+verified are executed as the system boots. This document describes the process
+by which OpenTitan aims to ensure this property holds, as well as the services
+that the secure boot chain provides to software, such as attestation and
+firmware upgrade support.
+
+# References
+
+*   [Attestation][attestation]
+*   [Ownership Transfer][ownership-transfer]
+*   [OpenTitan Secure Boot HW Support][secure-boot-hw-support]
+*   [OpenTitan Flash][ot-flash]
+
+# Terminology
+
+## RISC-V Concepts:
+
+*   **PMP:** Physical Memory Protection unit from the
+    [RISC-V ISA Volume II: Privileged Architecture](rv-isa-priv) specification.
+    Allows defining properties (RWX) for regions of physical memory.
+
+## Boot stages:
+
+*   `ROM`: Metal mask ROM, sometimes known as Boot ROM.
+*   `ROM_EXT`: ROM Extension. Stored in flash and signed by the Silicon Creator.
+*   `BL0`: Bootloader. Signed by the Silicon Owner.
+*   `Kernel`: Signed by the Silicon Owner.
+
+# Boot Process Overview
+
+The boot process flows as follows:
+
+1.  Power on.
+    1.  Execution is restricted to the ROM region via
+        [OpenTitan Secure Boot HW Support](secure-boot-hw-support).
+    2.  The last PMP region (region #15) is configured by reset logic to cover
+        the entire flash region, with only the R and L bits set (that is, the
+        region is **R**ead-only and **L**ocked, meaning the region is respected
+        in Machine mode and cannot be unlocked unless by a system reset).
+2.  Execution enters ROM stage.
+    1.  All SRAM except for retention SRAM is cleared.
+    2.  The active boot slot is loaded from the flash Boot Info. This value is
+        called the Active `ROM_EXT` Slot.
+    3.  Starting with the Active `ROM_EXT` Slot, the ROM code performs the
+        following:
+        1.  Determine if the slot is empty by testing for presence of the header
+            magic value.
+            1.  If present, continue validation. If missing, enter boot failure
+                logic.
+        2.  Compute the digest of the entire payload, as specified in the slot
+            header.
+            1.  If the digest doesn’t match the payload digest header field,
```

Comment:
nit: Should we remove this for now since we are rolling this into the signature verification? Timing wise I don't think we are saving much, so we can use the hash embedded in the signature if we want to check for flash corruption (assuming we use a padding scheme).

Reply:
Fair point, removed.